compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp

```diff
@@ -1097,6 +1097,23 @@ static LogicalResult setReductionConfig(const spirv::TargetEnv &targetEnv,
 // Everything Default Configuration
 //===----------------------------------------------------------------------===//
 
+/// Returns a small tiling factor for the given reduction `dimSize`.
+/// Returns 0 to avoid tiling.
+static int getReductionTilingFactor(int64_t dimSize) {
+  if (dimSize % 4 == 0) return 4;
+
+  // Try to find the smallest prime factor as the tiling factor. As a trade off
+  // between generated code size and compilation time, only look at prime
+  // numbers less than 50 right now.
+  const std::array<int, 15> primeNumbers = {2,  3,  5,  7,  11, 13, 17, 19,
```

Comment:
ubernit: `static constexpr` or drop const? IIRC the MLIR style discourages const on local variables